### ISSUE TITLE
NullReferenceException running .NET Core based xUnit tests in parallel when using AddCollectionMappers() and ServiceCollectionExtensions.AddAutoMapper() together

### ISSUE THREAD

**Issue by prochnowc** [AutoMapper/AutoMapper.Collection#49]
When running .NET Core based xUnit tests in parallel which use `AddCollectionMappers()` together with `ServiceCollectionExtensions.AddAutoMapper()` i receive sometimes `NullReferenceException`:

```
System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at AutoMapper.EquivalencyExpression.EquivalentExpressions.<>c__DisplayClass5_0`1.<InsertBefore>b__1(IConfigurationProvider c)
   at AutoMapper.MapperConfiguration..ctor(MapperConfigurationExpression configurationExpression)
   at AutoMapper.ServiceCollectionExtensions.AddAutoMapperClasses(IServiceCollection services, Action`1 additionalInitAction, IEnumerable`1 assembliesToScan)
   at AutoMapper.ServiceCollectionExtensions.AddAutoMapper(IServiceCollection services, Action`1 additionalInitAction, Type[] profileAssemblyMarkerTypes) 
```

I'm sorry but i cannot provide unit test to reproduce the issue - it highly depends on timing.
But looking at `AddAutoMapper` and `AddCollectionMappers` for me it seems to be caused by static initialization:

- `AddAutoMapper` internally uses `Mapper.Initialize` (statics are initialized multiple times)
- `AddCollectionMappers` internally uses static fields from `EquivalentExpressions` which are accessed conurrently during static Mapper.Initialize

So in my opinion we have two issues here:
- `AddAutoMapper` should not initialize static (as far as i know there is only one Process per parallel running unit tests in one collection). So in fact `Mapper.Initialize()` is called multiple times per process.
- `EquivalentExpressions` should use static concurrent dictionaries. 

Of course, both issues can be worked around disabling parallel unit test execution.

`[assembly: CollectionBehavior(DisableTestParallelization = true)]`

**Comment by jbogard** [comment]
Yeah, just turn parallelization off.

**Comment by TylerCarlson1** [comment]
I'm with @jbogard on this.  It causes many problems you wouldn't usually find if you make one mapper at start like you are suppose to.

That being said I wouldn't be opposed to a PR with this fix and a test for this scenario.

**Comment by Tasteful** [comment]
This can be closed, the NRE was fixed by #78

@TylerCarlson1
